### PR TITLE
Run resnext101 training in rocm benchmark

### PR DESCRIPTION
**Pull request by bddppq**
cc @xw285cornell

### CODE REVIEW

**Comment by bddppq** [comment]
ubuntu:
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/13305/
centos:
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-devtoolset7-rocmrpm-centos7.5-trigger-test/6299/

**Review by houseroad — APPROVED** [review]
Looks good.

**Comment by bddppq** [comment]
ubuntu:
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/13320/
centos:
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-devtoolset7-rocmrpm-centos7.5-trigger-test/6316/

File: .jenkins/caffe2/bench.sh [hunk 1]
```diff
@@ -13,11 +13,14 @@ else
...
 if (( $num_gpus == 0 )); then
-    cmd="$cmd --use_cpu"
+    "$PYTHON" "$CAFFE2_PYPATH/python/examples/resnet50_trainer.py" --train_data null --batch_size 64 --epoch_size 6400 --num_epochs 2 --use_cpu
 else
-    cmd="$cmd --num_gpus 1"
+    "$PYTHON" "$CAFFE2_PYPATH/python/examples/resnet50_trainer.py" --train_data null --batch_size 64 --epoch_size 6400 --num_epochs 2 --num_gpus 1
+    if (( $num_gpus >= 2 )); then
+        "$PYTHON" "$CAFFE2_PYPATH/python/examples/resnet50_trainer.py" --train_data null --batch_size 128 --epoch_size 12800 --num_epochs 2 --num_gpus 2
+    fi
+    if (( $num_gpus >= 4 )); then
```
Comment (houseroad) [inline .jenkins/caffe2/bench.sh]: Do we want to test both 2 and 4 on the machines with 4+ gpus?

Comment (bddppq) [reply .jenkins/caffe2/bench.sh]: Why not? :-)

**Comment by bddppq** [comment]
ubuntu:
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/13322/
centos:
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-devtoolset7-rocmrpm-centos7.5-trigger-test/6316/

**Comment by petrex** [comment]
@bddppq  do you guys have the new vbios on CI nodes?

File: .jenkins/caffe2/bench.sh [hunk 2]
```diff
@@ -13,11 +13,30 @@ else
...
+    "$PYTHON" "$CAFFE2_PYPATH/python/examples/resnet50_trainer.py" --train_data null --batch_size 256 --epoch_size 25600 --num_epochs 2 --num_gpus 4
 fi
 
-eval "$cmd"
+# ResNext
+if (( $num_gpus == 0 )); then
+    "$PYTHON" "$CAFFE2_PYPATH/python/examples/resnet50_trainer.py" --resnext_num_groups 32 --resnext_width_per_group 4 --train_data null --batch_size 32 --epoch_size 3200 --num_epochs 2 --use_cpu
+fi
+if (( $num_gpus >= 1 )); then
+    "$PYTHON" "$CAFFE2_PYPATH/python/examples/resnet50_trainer.py" --resnext_num_groups 32 --resnext_width_per_group 4 --train_data null --batch_size 32 --epoch_size 3200 --num_epochs 2 --num_gpus 1
```
Comment (xw285cornell) [inline .jenkins/caffe2/bench.sh]: Add `--num_layers 101` maybe?

**Review by xw285cornell — APPROVED** [review]
LGTM! 

hmm it looks there is a seg fault on the 4 GPU run: https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-bench/323/console

Seems like a concurrency issue for the HSAQueue?

**Comment by bddppq** [comment]
@petrex yes bench runs are running on worker 28 which has got updated vbios.

**Comment by bddppq** [comment]
@petrex @ashishfarmer Could you guys help on the 4-gpu segfault @xw285cornell mentioned above? 
```
21:16:41 INFO:ResNe(X)t_trainer:Training loss: 0.0, accuracy: 1.0
21:16:42 *** Aborted at 1547500602 (unix time) try "date -d @1547500602" if you are using GNU date ***
21:16:42 PC: @     0x7fa6ecd7b919 Kalmar::HSAQueue::isEmpty()
21:16:42 *** SIGSEGV (@0x0) received by PID 9100 (TID 0x7fa63af51700) from PID 0; stack trace: ***
21:16:42     @     0x7fa7cda6d390 (unknown)
21:16:42     @     0x7fa6ecd7b919 Kalmar::HSAQueue::isEmpty()
21:16:42     @     0x7fa6ecd6ab2d Kalmar::HSADevice::createOrstealRocrQueue()
21:16:42     @     0x7fa6ecd6d3b2 Kalmar::HSAQueue::acquireLockedRocrQueue()
21:16:42     @     0x7fa6ecd97549 HSABarrier::enqueueAsync()
21:16:42     @     0x7fa6ecd7d1ca Kalmar::HSAQueue::EnqueueMarkerWithDependency()
21:16:42     @     0x7fa7c12d0658 hc::accelerator_view::create_marker()
21:16:42     @     0x7fa7c12d056c ihipStream_t::locked_recordEvent()
21:16:42     @     0x7fa7c134d02e hipEventRecord
21:16:42     @     0x7fa708a4427a caffe2::EventRecordHIP()
21:16:42     @     0x7fa708a47b35 caffe2::Event::Record()
21:16:42     @     0x7fa708a46b0d caffe2::Operator<>::RunAsync()
21:16:42     @     0x7fa798a6658f caffe2::AsyncNetBase::run()
21:16:42     @     0x7fa798a6c49a caffe2::AsyncSchedulingNet::schedule()::$_0::operator()()
21:16:42     @     0x7fa79781f1f5 c10::ThreadPool::main_loop()
21:16:42     @     0x7fa7299a4c80 (unknown)
21:16:42     @     0x7fa7cda636ba start_thread
21:16:42     @     0x7fa7cd79941d clone
21:16:42     @                0x0 (unknown)
```

File: .jenkins/caffe2/bench.sh (hunk 2 above)
Comment (bddppq) [reply .jenkins/caffe2/bench.sh]: Added

**Comment by bddppq** [comment]
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/15904/
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-devtoolset7-rocmrpm-centos7.5-trigger-test/8906/

**Pull request closed without merging**